### PR TITLE
Add support for pre-focus lens in FWHM calculation

### PR DESCRIPTION
**Pull request by vespos**
## Description
Add option to use pre-focusing lens when calculating the FWHM of the beam

## Motivation and Context
Lens calculation in the FEH is quite off when not accounting for the pre-focussing lens set.

## How Has This Been Tested?
Used at XCS in June.
Added a test instance for this, based on the parameters used in the hutch

## Where Has This Been Documented?
Docstring

### CODE REVIEW

**Comment by vespos** [comment]
There is an issue with xraylib for the build. I can look into that but if you already know a solution, let me know @ZLLentz

**Comment by ZLLentz** [comment]
xraylib is not compatible with numpy 2, which released recently. The conda builds know this and set version accordingly, the pip builds do not. See the small edit we made in pcdsdevices to add numpy<2 as a CI-time constraint: https://github.com/pcdshub/pcdsdevices/blob/c568a607c3fc6e998a158874150b5a561dc16aac/.github/workflows/standard.yml#L24

**Comment by ZLLentz** [comment]
d'oh, check out #29 which I'll merge now, no need to adjust anything here

**Comment by ZLLentz** [comment]
I am going to click "update branch", note that you will need to pull this update back to your clone to make more pushes

File: pcdscalc/tests/test_be_lens_calcs.py [hunk 1]
```diff
@@ -225,6 +225,28 @@ def test_calc_beam_fwhm_with_source_distance(energy_sample, lens_set, dist,
...
+    fwhm = be_lens_calcs.calc_beam_fwhm(energy=energy_sample,
+                                        lens_set=lens_set,
+                                        distance=dist,
+                                        fwhm_unfocused=fwhm_unf,
+                                        source_distance=source_dist,
+                                        prefocus_set=prefocus_set,
+                                        prefocus_distance=prefocus_distance
+                                        )
+    logger.debug('Expected: %s, Received: %s', expected, fwhm)
+    assert np.isclose(fwhm, expected)
```
Comment (ZLLentz) [inline pcdscalc/tests/test_be_lens_calcs.py]: The only obvious thing to (optionally) improve here is to combine this new fwhm test with the old test. The whole point of `@pytest.mark.parametrize` is to re-use the same test with multiple inputs and outputs so this seems wasteful.

**Review by ZLLentz — DISMISSED** [review]
Functionally and code-health-wise I have no complaints, feel free to merge!

File: pcdscalc/tests/test_be_lens_calcs.py (hunk 1 above)
Comment (vespos) [reply pcdscalc/tests/test_be_lens_calcs.py]: Yeah I can do that!

**Comment by vespos** [comment]
Hopefully xraylib will be numpy2 compatible sson: https://github.com/tschoonj/xraylib/pull/240

**Comment by vespos** [comment]
I have combined all calc_beam_fwhm tests into a single function, let me know if you think there's something else I could do!

**Review by ZLLentz — APPROVED** [review]
LGTM

**Pull request merged**